### ISSUE TITLE
Carousel works only after the page is resized 

### ISSUE THREAD

**Issue by joshuadsouza08** [ismail9k/vue3-carousel#134]
The carousel displays the images when mounted but does not go to the next image unless the page is resized and this happens only if there is a 'd-flex' bootstrap property in the parent Div element, without the 'd-flex' the carousel works fine. is there any reason for this?

**Comment by thalida** [comment]
I ran into a similar issue as joshua today when creating an image carousel, on load the carousel wasn't scrolling fully to the next page until I resized the window (see bug gif below). In my case `updateSlideWidth` is called before the images load, so it gets the default slide width of ~200px. The easiest solution for me was to update my css to have a fixed width. 

I'm curious what your template and css looks like @joshuadsouza08? 
(I commented on this issue instead of creating a new one since it's a similar outcome "the carousel only works after the page is resized", but might be caused for different reasons, and may require a new issue depending on how ismail9k wants to do tracking. The solution for mine could be adding an image carousel example to the documentation, instead of a logic change.)  

**Template / CSS To Reproduce:**
template:
```html
<div class="gallery">
    <carousel>
      <slide v-for="photo in photos as any[]" :key="photo.id">
        <div class="carousel__item">
          <img :src="photo.image_url" alt="" />
        </div>
      </slide>
      <template #addons>
        <navigation />
      </template>
    </carousel>
</div>
```

original css:
```css
.carousel__item {
  min-height: 200px;
  width: 100%;
  background-color: var(--vc-clr-primary);
  color: var(--vc-clr-white);
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;

  img {
    height: 100%;
  }
}
```
bug gif:
![before](https://user-images.githubusercontent.com/3401715/145106478-3ef30704-d07f-4ecf-8ff4-a02ee59861df.gif)


**Solution**
(same template)
Updated css:
```css
.carousel__item {
  height: 500px;
  width: 500px; // instead of 100%
  background-color: var(--vc-clr-primary);
  color: var(--vc-clr-white);
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;

  img {
    width: 100%; // constrain image width
  }
}
```
fixed gif:
![after](https://user-images.githubusercontent.com/3401715/145106498-c6b5001e-359d-4885-b4d5-b16b8d8aaaf6.gif)

**Comment by joshuadsouza08** [comment]
i am rendering an adaptive card inside a 'div' element of the <Slide> tag.When I add the css property d-flex to the parent div element i get the bug as shown in the bug gif above.

`<template>
  <div class="msg" id="messageclass">
    <div v-if="message.hasAdaptiveCard || message.hasMarkDown" :id="message.id">
      <div v-if="message.hascarousel">
        <Carousel :items-to-show="1" :itemsToScroll="1" :transition="200">
          <Slide
            v-for="slide in message.innerHtml.length"
            :id="`carousel_${message.id + slide}`"
            :key="slide"
            class="carosuel__item"
            ><div
              :id="`carouseldiv_${message.id + slide}`"
              class="carousel__div"
            ></div>
          </Slide>

          <template #addons>
            <Navigation />
            <Pagination />
          </template>
        </Carousel>
      </div>
    </div>
    <div v-else>
      {{ message.text }}
    </div>
  </div>
</template>
<script>
import { Carousel, Navigation, Pagination, Slide } from "vue3-carousel";

import "vue3-carousel/dist/carousel.css";
export default {
  name: "message-content",
  props: {
    message: {
      type: Object,
      default: null,
    },

  },

  components: {
    Carousel,
    Slide,
    Pagination,
    Navigation,
  },

  mounted() {
    if (
      document.getElementById(this.message.id) != null &&
      this.message.innerHtml != ""
    ) {
      if (this.message.hascarousel) {
        for (let i = 0; i < this.message.innerHtml.length; i++) {
          let slideCount = i + 1;
          let divelement = document.getElementById(
            `carouseldiv_${this.message.id + slideCount}`
          );
          var test = document
            .getElementById(`carousel_${this.message.id + slideCount}`)
            .appendChild(divelement);
          document
            .getElementById(`carouseldiv_${this.message.id + slideCount}`)
            .appendChild(this.message.innerHtml[i]);
          console.log(test);
        }
      } else {
        document
          .getElementById(this.message.id)
          .appendChild(this.message.innerHtml);
      }
    }
  },
};
</script>
`

i tried using the css you suggested @thalida but it did not work for me. am i doing something wrong here ?

**Comment by lacorde** [comment]
Hi @joshuadsouza08, I have the same exact behaviour as yours.

Using [the basic example](https://github.com/ismail9k/vue3-carousel) provided by @ismail9k, and no specific CSS.

Carousel is working fine when the app is first mounted. Then if I change page (I'm using Nuxt), and then go back to the page including the slider, navigation/slides stop working.

**Comment by ismail9k** [comment]
I will look into this issue. For now, as a temporary fix for this issue, you can use `restartCarousel()` method to recalculate the carousel slide width.

**Comment by lacorde** [comment]
@ismail9k Thank you for your answer.
In the end, the following code worked for me (setting the initial slide on mounted)  @joshuadsouza08 

```
<template>
  <carousel :items-to-show="1.5" v-model="initialSlide">
    <slide v-for="slide in 10" :key="slide">
      {{ slide }}
    </slide>
    <template #addons>
      <navigation />
      <pagination />
    </template>
  </carousel>
</template>

<script>
export default {
 data() {
  return {
   initialSlide: undefined,
  };
 },
 mounted() {
  this.initialSlide = 0;
 },
{
</script>
```
